playbooks/common/openshift-cluster/redeploy-certificates.yml

```diff
@@ -175,25 +181,63 @@
   - name: restart node
     service: name={{ openshift.common.service_type }}-node state=restarted
 
-- name: Redeploy OpenShift Registry
+- name: Copy admin client config(s)
   hosts: oo_first_master
   tasks:
   - name: Create temp directory for kubeconfig
     command: mktemp -d /tmp/openshift-ansible-XXXXXX
     register: mktemp
     changed_when: False
-  - name: Copy the admin client config(s)
+
+  - name: Copy admin client config(s)
     command: >
       cp {{ openshift.common.config_base }}/master//admin.kubeconfig {{ mktemp.stdout }}/admin.kubeconfig
     changed_when: False
-  - name: Determine if registry deployed
-    command: "oc --config={{ mktemp.stdout }}/admin.kubeconfig get dc/docker-registry -n default"
-    register: docker_registry
+
+- name: Serially evacuate all nodes to trigger redeployments
+  hosts: oo_nodes_to_config
+  serial: 1
+  any_errors_fatal: true
+  tasks:
+  - name: Determine if node is currently scheduleable
+    command: >
+      {{ openshift.common.client_binary }} --config={{ mktemp.stdout }}/admin.kubeconfig
+      get node {{ openshift.common.hostname | lower }} -o json
+    register: node_output
+    when: openshift_certificates_redeploy_ca | default(false) | bool
+    delegate_to: "{{ groups.oo_first_master.0 }}"
     changed_when: false
-    failed_when: false
-  - name: Redeploy registry
-    command: "oc --config={{ mktemp.stdout }}/admin.kubeconfig deploy dc/docker-registry --latest -n default"
-    when: docker_registry.rc == 0
+
+  - set_fact:
+      was_schedulable: "{{ 'unschedulable' not in (node_output.stdout | from_json).spec }}"
+    when: openshift_certificates_redeploy_ca | default(false) | bool
+
+  - name: Prepare for node evacuation
+    command: >
+      {{ openshift.common.admin_binary }} --config={{ mktemp.stdout }}/admin.kubeconfig
+      manage-node {{ openshift.common.hostname | lower }}
+      --schedulable=false
+    delegate_to: "{{ groups.oo_first_master.0 }}"
+    when: openshift_certificates_redeploy_ca | default(false) | bool and was_schedulable | bool
+
+  - name: Evacuate node
+    command: >
+      {{ openshift.common.admin_binary }} --config={{ mktemp.stdout }}/admin.kubeconfig
+      manage-node {{ openshift.common.hostname | lower }}
+      --evacuate --force
+    delegate_to: "{{ groups.oo_first_master.0 }}"
+    when: openshift_certificates_redeploy_ca | default(false) | bool and was_schedulable | bool
+
```

Comment:
My understanding was that node evac blocks until completed. Checked with whearn on IRC and sounds like nothing we have to worry about, go for it.